lib/ansible/modules/network/cloudengine/ce_ntp_auth.py

```diff
@@ -0,0 +1,588 @@
+        trusted_key = self.trusted_key
+        if self.state == 'absent':
+            auth_type = None
+            trusted_key = None
+        self.proposed = dict(key_id=self.key_id, auth_pwd=self.password,
+                             auth_mode=self.auth_mode, auth_type=auth_type,
+                             trusted_key=trusted_key, authentication=self.authentication,
+                             state=self.state)
+
+    def get_update_cmd(self):
+        """Get updated commands"""
+
+        cli_str = ""
+        if self.state == "present":
+            cli_str = "ntp authentication-keyid %s authentication-mode %s " % (
+                self.key_id, self.auth_mode)
+            if self.auth_type == 'encrypt':
+                cli_str = "%s cipher %s" % (cli_str, self.password)
+            else:
+                cli_str = "%s %s" % (cli_str, self.password)
+        else:
+            cli_str = "undo ntp authentication-keyid %s" % self.key_id
+
+        self.updates_cmd.append(cli_str)
+
+        if self.authentication:
+            cli_str = ""
+
+            if self.ntp_auth_conf['authentication'] != self.authentication:
+                if self.authentication == 'enable':
+                    cli_str = "ntp authentication enable"
+                else:
+                    cli_str = "undo ntp authentication enable"
+
+            if cli_str != "":
+                self.updates_cmd.append(cli_str)
+
+        cli_str = ""
+        if self.state == "present":
+            if self.trusted_key != self.cur_trusted_key:
+                if self.trusted_key == 'enable':
+                    cli_str = "ntp trusted authentication-keyid %s" % self.key_id
+                else:
+                    cli_str = "undo ntp trusted authentication-keyid %s" % self.key_id
+        else:
+            cli_str = "undo ntp trusted authentication-keyid %s" % self.key_id
+
+        if cli_str != "":
+            self.updates_cmd.append(cli_str)
+
+    def get_end_state(self):
+        """Get end state info"""
+
+        self.ntp_auth_conf = dict()
+        self.get_ntp_auth_exist_config()
+        self.end_state = copy.deepcopy(self.ntp_auth_conf)
+
+    def show_result(self):
+        """Show result"""
+
+        self.results['changed'] = self.changed
+        self.results['proposed'] = self.proposed
+        self.results['existing'] = self.existing
+        self.results['end_state'] = self.end_state
+        if self.changed:
+            self.results['updates'] = self.updates_cmd
+        else:
+            self.results['updates'] = list()
+
+        self.module.exit_json(**self.results)
+
+    def work(self):
+        """Excute task"""
+
+        self.get_existing()
+        self.get_proposed()
+        self.get_update_cmd()
+
+        self.config_ntp_auth()
+
+        self.get_end_state()
+        self.show_result()
+
+
+def main():
+    """Main function entry"""
+
+    argument_spec = dict(
+        key_id=dict(required=True, type='str'),
```

Comment:
Because the device side configuration is all type ```string```, avoid repeated conversion, Let's keep it.